api/src/main/resources/sql/procedures/create_functions.sql

```diff
@@ -907,3 +907,75 @@ RETURN ret;
 END
 
 #
+
+-- This function accepts encounter_id, mapping source, mapping code, offset value
+-- It will find a single, best observation that matches this, and return the value_datetime
+-- usage obs_value_datetime_by_offset(encounter_id, 'CIEL or PIH', "1234 OR TEXT");
+#
+DROP FUNCTION IF EXISTS obs_value_datetime_by_offset;
+#
+CREATE FUNCTION obs_value_datetime_by_offset(_encounterId int(11), _source varchar(50), _term varchar(255), _offset_value int(11))
+RETURNS datetime
+DETERMINISTIC
+
+BEGIN
+
+DECLARE ret datetime;
+
+select      o.value_datetime into ret
+from        obs o
+where       o.voided = 0
+and         o.encounter_id = _encounterId
+and         o.concept_id = concept_from_mapping(_source, _term)
+order by    o.obs_id asc
+limit 1
+offset _offset_value;
+
+RETURN ret;
+
+END
+
+#
+
+
+-- This function accepts encounter_id, mapping source, mapping code
+-- It will find a single, best observation that matches this, group them by the obs_group_id and return the value_text
```

Comment:
These comments need to be updated to reflect what is happening.

Reply:
I really don't know what this function does by reading it, and the comments aren't helpful in understanding.  Please document here what this function is intending to do and when you'd choose to use it.